src/objective-c/GRPCClient/private/GRPCSecureChannel.m

```diff
@@ -47,14 +59,34 @@ - (instancetype)initWithHost:(NSString *)host {
     NSAssert(certsPath.length,
              @"gRPCCertificates.bundle/roots.pem not found under %@. This file, with the root "
              "certificates, is needed to establish TLS (HTTPS) connections.", bundle.bundlePath);
-    NSData *certsData = [NSData dataWithContentsOfFile:certsPath];
-    NSAssert(certsData.length, @"No data read from %@", certsPath);
-    NSString *certsString = [[NSString alloc] initWithData:certsData encoding:NSUTF8StringEncoding];
-    kCredentials = grpc_ssl_credentials_create(certsString.UTF8String, NULL);
+    kDefaultCertificates = CertificatesAtPath(certsPath);
   });
-  return (self = [super initWithChannel:grpc_secure_channel_create(kCredentials,
-                                                                   host.UTF8String,
-                                                                   NULL)]);
+  grpc_credentials *certificates = path ? CertificatesAtPath(path) : kDefaultCertificates;
+
+  // Ritual to pass the SSL host name override to the C library.
```

Comment:
We might need other channel arguments at some point. Maybe it would be better to have a more general solution.